File: models/drop_party.py
from datetime import datetime, timezone
from sqlmodel import Field, SQLModel, Relationship  # ty:ignore[unresolved-import]
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .drop import Drop


class Drop_Party(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    world: int
    avg_drop: int
    is_active: bool = Field(default=True)
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    last_drop_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    drops: list["Drop"] = Relationship(back_populates="drop_party")

File: src/main/java/rip/jade/partytimeserverjava/entity/DropParty.java
package rip.jade.partytimeserverjava.entity;// Drop.java

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "drop_party")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DropParty {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Integer world;

    @Column(name = "avg_drop", nullable = false)
    private Integer avgDrop;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "last_drop_at", nullable = false)
    @Builder.Default
    private Instant lastDropAt = Instant.now();

    @OneToMany(mappedBy = "dropParty", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Drop> drops = new ArrayList<>();

    // Helper methods for bidirectional relationship
    public void addDrop(Drop drop) {
        drops.add(drop);
        drop.setDropParty(this);
    }

    public void removeDrop(Drop drop) {
        drops.remove(drop);
        drop.setDropParty(null);
    }
}